app/models/user.rb

```diff
@@ -88,8 +95,14 @@ class User < ActiveRecord::Base
   after_save :clear_global_notice_if_needed
   after_save :refresh_avatar
   after_save :badge_grant
+  after_save do
+    DiscourseEvent.trigger(:user_verified, user: self) if (self.active_changed? && self.active == true)
+  end
 
   before_destroy do
+    # Event hook
+    DiscourseEvent.trigger(:user_deleted, user: self)
```

Comment:
Good question.

Would it be better to move the trigger to after_destroy..?

Or better yet, rename the trigger to before_user_deleted and create another trigger called after_user_deleted?
That way you can play around with the logic if the user was actually deleted or not .. (in case you need to access the user object before `destroy`)

``` ruby
before_destroy do
  DiscourseEvent.trigger(:before_user_deleted, user: self)
  ...
end
after_destroy do
  DiscourseEvent.trigger(:after_user_deleted, user_id: self.id, self.email, self.username, self.name)
end
```